### PR TITLE
CI + bunch of fixes

### LINKED ISSUE #40: Add circle CI configuration
I noticed some test files are missing, we need to add circle ci support to make sure each commit doesn't leave anything out.

### PR DESCRIPTION
**Pull request by AkihiroSuda**
#### `.travis.yml`:
- Newly added (Fix #40, let me know if you prefer other CI :smile:)

#### `.mailmap`, `AUTHORS`, `Makefile`:
- Fix `make AUTHORS`, which was broken because of lack of `.mailmap` (I can remove this target if unneeded)
- Skip running lint by default because it is too noisy about "comments are missing"
- Remove wrong `${DOCKER_BUILDTAGS}`
- Fix `make binaries`, which was just an empty target

#### `README.md`: 
- Add godoc badge and travis badge

#### `digests_test.go`, `resource.go`: 
- Fix for `go vet`

#### `manifest_test.go`, `resource_test.go`, `testutil_test.go`:
- Add missing `diffResourceList()` from #39 (@dmcgowan)
- Remove dependency for `tree(1)`
- Fix for `go vet`
- Fix the following test failure;
```
manifest_test.go:163: Resource list difference
manifest_test.go:171: Changed resource:
                    Expected: &continuity.symLink{resource:continuity.resource{paths:[]string{"/c/a-abssymlink"}, mode:0x80001ff, uid:"1001", gid:"100
1", xattrs:map[string][]uint8(nil)}, target:"/a"}
                    Actual:   &continuity.symLink{resource:continuity.resource{paths:[]string{"/c/a-abssymlink"}, mode:0x80001ff, uid:"1001", gid:"100
1", xattrs:map[string][]uint8{}}, target:"/tmp/continuity-test-356664233/a"}
    
```

Now CI passing: https://travis-ci.org/AkihiroSuda/continuity/builds/208498783

I'm also planning to do vendor using vndr in another PR.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -45,14 +46,15 @@ lint:
...
+	@go build -v ${GO_LDFLAGS} ./...
 
 test:
 	@echo "+ $@"
-	@go test -tags "${DOCKER_BUILDTAGS}" ./...
+	@go test ./...
 
 binaries: ${PREFIX}/bin/continuity
 	@echo "+ $@"
+	@go build -o $< ${GO_LDFLAGS} ./cmd/continuity
```
Comment (dmcgowan) [inline Makefile]: This line doesn't need to be here, `${PREFIX}/bin/continuity` already causes this to be built.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+language: go
+
+go:
+  - 1.8
```
Comment (dmcgowan) [inline .travis.yml]: Lets build 1.7.x here as well

**Comment by dmcgowan** [comment]
Thanks for getting all this in shape! Just a couple small changes on the build process and we can get this merged.

**Comment by AkihiroSuda** [comment]
done @dmcgowan

**Comment by dmcgowan** [comment]
LGTM

I think we may want to move some of the test functions around in the future. Maybe to a subpackage so they can be imported by other packages.

**Comment by dmcgowan** [comment]
@stevvooe going to merge this, if you want further changes we can do in follow up

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+language: go
+
+go:
+  - 1.7
+  - 1.8
```
Comment (stevvooe) [inline .travis.yml]: tip, pls.

File: AUTHORS [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Aaron Lehmann <aaron.lehmann@docker.com>
+Akihiro Suda <suda.akihiro@lab.ntt.co.jp>
+Brandon Philips <brandon.philips@coreos.com>
+Derek McGowan <derek@mcgstyle.net>
+Stephen Day <stevvooe@users.noreply.github.com>
```
Comment (stevvooe) [inline AUTHORS]: Please make an entry in the mail map for this address.

**Pull request merged**